reconcile/sql_query.py

```diff
@@ -385,112 +377,151 @@ def process_template(query, image_repository, use_pull_secret=False):
+    return f"cat ''{directory}''/{file_glob} > ''{output_file}'';"
+
+
+def openshift_apply(
+    dry_run: bool,
+    oc_map: OC_Map,
+    cluster: str,
+    namespace: str,
+    resources: Iterable[OpenshiftResource],
+) -> None:
+    for resource in resources:
+        openshift_base.apply(
             dry_run=dry_run,
             oc_map=oc_map,
-            cluster=query["cluster"],
-            namespace=query["namespace"]["name"],
-            resource_type=resource_type,
-            name=query["name"],
-            enable_deletion=enable_deletion,
-        )
-    except StatusCodeError:
-        LOG.exception(
-            "Error removing ['%s' '%s' '%s' '%s']",
-            query["cluster"],
-            query["namespace"]["name"],
-            resource_type,
-            query["name"],
+            cluster=cluster,
+            namespace=namespace,
+            resource_type=resource.kind,
+            resource=resource,
+            wait_for_namespace=False,
         )
 
 
-def run(dry_run, enable_deletion=False):
+def openshift_delete(
+    dry_run: bool,
+    oc_map: OC_Map,
+    cluster: str,
+    namespace: str,
+    resources: Iterable[OpenshiftResource],
+    enable_deletion: bool,
+) -> None:
+    for resource in resources:
+        try:
+            openshift_base.delete(
+                dry_run=dry_run,
+                oc_map=oc_map,
+                cluster=cluster,
+                namespace=namespace,
+                resource_type=resource.kind,
+                name=resource.name,
+                enable_deletion=enable_deletion,
+            )
+        except StatusCodeError:
+            LOG.exception(
+                f"Error removing ['{cluster}' '{namespace}' '{resource.kind}' '{resource.name}']"
+            )
+
+
+def run(dry_run: bool, enable_deletion: bool = False) -> None:
     settings = queries.get_app_interface_settings()
     accounts = queries.get_state_aws_accounts()
     state = State(
         integration=QONTRACT_INTEGRATION, accounts=accounts, settings=settings
     )
+    image_repository = "quay.io/app-sre"
+
+    sql_query_settings = settings.get("sqlQuery")
+    pull_secret: dict[str, Any] = {}
+    if sql_query_settings:
+        image_repository = sql_query_settings["imageRepository"]
+        pull_secret = sql_query_settings["pullSecret"]
+    use_pull_secret = True if pull_secret else False
 
     queries_list = collect_queries(settings=settings)
-    remove_candidates = []
     for query in queries_list:
+        openshift_resources: list[OpenshiftResource] = []
         query_name = query["name"]
+        cleanup = False
 
         # Checking the sql-query state:
         # - No state: up for execution.
         # - State is a timestamp: executed and up for removal
         #   after the JOB_TTL
         # - State is 'DONE': executed and removed.
         # - State is not 'DONE' but 'delete:true' and is a cronjob: up for removal
-        try:
+        if state.exists(query_name):
```

Comment:
Do we really need all this state handling refactor for the configmap split? Since I'm not very familiar with the original integration I'm having a hard time following all the changes here 😓 

I understand that the current state logic is a bit convoluted and I would agree in attempting a refactor outside this PR 😄